src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -0,0 +1,579 @@
+            get { return Interop.Termios.TermiosGetAvailableBytes(_handle, Interop.Termios.Queue.SendQueue); }
+        }
+
+        internal int BytesToRead
+        {
+            get { return Interop.Termios.TermiosGetAvailableBytes(_handle, Interop.Termios.Queue.ReceiveQueue); }
+        }
+
+        internal bool CDHolding
+        {
+            get
+            {
+                int status = Interop.Termios.TermiosGetSignal(_handle, Interop.Termios.Signals.SignalDcd);
+                if (status < 0)
+                {
+                    throw new IOException();
+                }
+
+                return status == 1;
+            }
+        }
+
+        internal bool CtsHolding
+        {
+            get
+            {
+                int status = Interop.Termios.TermiosGetSignal(_handle, Interop.Termios.Signals.SignalCts);
+                if (status < 0)
+                {
+                    throw new IOException();
+                }
+
+                return status == 1;
+            }
+        }
+
+        internal bool DsrHolding
+        {
+            get
+            {
+                int status = Interop.Termios.TermiosGetSignal(_handle, Interop.Termios.Signals.SignalDsr);
+                if (status < 0)
+                {
+                    throw new IOException();
+                }
+
+                return status == 1;
+            }
+        }
+
+        internal bool DtrEnable
+        {
+            get
+            {
+                int status = Interop.Termios.TermiosGetSignal(_handle, Interop.Termios.Signals.SignalDtr);
+                if (status < 0)
+                {
+                    throw new IOException();
+                }
+
+                return status == 1;
+            }
+
+            set
+            {
+                if (Interop.Termios.TermiosGetSignal(_handle, Interop.Termios.Signals.SignalDtr, value ? 1 : 0) != 0)
+                {
+                    throw new IOException();
+                }
+            }
+        }
+
+        internal bool RtsEnable
+        {
+            get
+            {
+                int status = Interop.Termios.TermiosGetSignal(_handle, Interop.Termios.Signals.SignalRts);
+                if (status < 0)
+                {
+                    throw new IOException();
+                }
+
+                return status == 1;
+            }
+
+            set
+            {
+                if (Interop.Termios.TermiosGetSignal(_handle, Interop.Termios.Signals.SignalRts, value ? 1 : 0) != 0)
+                {
+                    throw new IOException();
+                }
+            }
+        }
+
+        internal Handshake Handshake
+        {
+            set
+            {
+                Debug.Assert(!(value < Handshake.None || value > Handshake.RequestToSendXOnXOff),
+                    "An invalid value was passed to Handshake");
+
+                if (value != _handshake)
+                {
+                    _handshake = value;
```

Comment:
This should be done only in case of success.

Reply:
Or perhaps value not cached at all since this is hardware and someone might want to ensure that the operation is actually done.